hibernate-core/src/main/java/org/hibernate/id/enhanced/SequenceStyleGenerator.java

```diff
@@ -216,7 +217,8 @@ public void configure(Type type, Properties parameters, ServiceRegistry serviceR
 				physicalSequence,
 				optimizationStrategy,
 				serviceRegistry,
-				determineContributor( parameters )
+				determineContributor( parameters ),
+				(ObjectNameNormalizer) parameters.get( IDENTIFIER_NORMALIZER )
```

Comment:
@mbladel `IDENTIFIER_NORMALIZER` was always a hack and has been deprecated for removal in H7. (It's no longer used there.)

We need a different approach here.

Reply:
I think you can use `jdbcEnvironment.getIdentifierHelper()` like in the H7 version of `IncrementGenerator`.

Reply:
I know it was a hack, but seeing it in different parts of the code I thought we could use that for now. I was not aware it was removed in H7.

 `IdentifierHelper` would indeed be available here, but would not expose the `PhysicalNamingStrategy` we need to get the database-side sequence name AFAIU.

Reply:
I think we need to make the `PhysicalNamingStrategy` available via the `IdentifierHelper` or something.

Right now, it's always obtained via the `Database` object, but I really don't see why that's a necessary/sensible/elegant design. 

In principle, if I'm not mistaken, anything with access to the `ServiceRegistry` could obtain the `PhysicalNamingStrategy` with a couple of lines of code.

Reply:
> I think we need to make the PhysicalNamingStrategy available via the IdentifierHelper or something.

Maybe make it available in `JdbcEnvironment`? It seems pertinent enough - though I'm not sure it's already available when instantiating it.

> In principle, if I'm not mistaken, anything with access to the ServiceRegistry could obtain the PhysicalNamingStrategy with a couple of lines of code.

You mean now, or after adding it to `IdentifierHelper`/`JdbcEnvironment`? Because I couldn't find a way with the current available services, unless one of them exposes the `Database` instance in some way.

Reply:
> Maybe make it available in `JdbcEnvironment`?

It's not quite perfect but I think it would be fine.

> You mean now

Well, I suppose I'm wrong. You can get a `PhysicalNamingStrategy` using:

```java
			final StrategySelector strategySelector = serviceRegistry.requireService( StrategySelector.class );
			final ConfigurationService configService = serviceRegistry.requireService( ConfigurationService.class );
			this.physicalNamingStrategy = strategySelector.resolveDefaultableStrategy(
					PhysicalNamingStrategy.class,
					configService.getSettings().get( AvailableSettings.PHYSICAL_NAMING_STRATEGY ),
					PhysicalNamingStrategyStandardImpl.INSTANCE
			);
```

But then I guess that doesn't respect one set explicitly by calling `MetadataBuilder.applyPhysicalNamingStrategy()`.

So I guess only code with access to the `MetadataBuilder` can obtain the `PhysicalNamingStrategy`.

Reply:
> > Maybe make it available in `JdbcEnvironment`?
> 
> It's not quite perfect but I think it would be fine.

I checked once again but sadly, I don't think this is possible: the `JdbcEnvironmentInitiator` doesn't have access to `MetadataBuildingOptions`, where the naming strategy is available, and is initialized before the `Database` instance is even created.

I spent quite a bit of time trying to find an alternative to this "hack" when fixing the bug without any success, and even now I don't see how we could access `Database` or `PhysicalNamingStrategy` without changing the signature of the `configure()` method (other than another hack-ish parameter).

Reply:
> changing the signature of the `configure()` method

Well then let's do that. (Overloading it with `default` for backward compatibility.)

Reply:
Alternatively, would it be better to do this in the calling code that sets the parameter, instead of doing it inside `SequenceStyleGenerator`?

Reply:
> do this in the calling code that sets the parameter

What do you mean by "do this"?

Reply:
> What do you mean by "do this"?

Call the naming strategy.

Reply:
That might work when we have an explicitly configured `sequence_name` parameter, but when the `SequenceStyleGenerator` derives it implicitly we would still need to go through the `PhysicalNamingStrategy` to be 100% correct I believe.

Note that also changing the `configure()` signature wouldn't be enough, there are many code paths which invoke that method and some of them don't have access to either the strategy or the `Database` object, so it might be a more intrusive change than I thought.

Reply:
Well the other _very_ reasonable option is to simply not fix this in H6.

Reply:
These things are only needed at bootstrap.  Contracts like JdbcEnvironment exist after bootstrap.  This is why they were currently accessible from Database.  You may not like it there, whatever, but at least scoping wise it makes much more sense.

Reply:
Let's avoid changing the contract of configure for 6.6.  I hate that method with a passion, but we need to be smart about hooking these generators into the better config approach.

Reply:
> Well the other very reasonable option is to simply not fix this in H6.

I totally get this if the fix needs major changes.  But here its totally fixable with a minor hack using something that's going away anyway.  The larger problem imo is `#configure` itself.